### PR TITLE
vktrace: fix error in vkUpdateDescriptorSets trim tracking

### PR DESCRIPTION
**Pull request by mizhen**
The current process cannot handle multiple bindings update in pDescriptorWrites,
the change fix the error.

XCAP-759
Change-Id: I4ba297e1171c652b59527d405bfba60e2be7a9a1

### CODE REVIEW

File: vktrace/vktrace_layer/vktrace_lib_trace.cpp [hunk 1]
```diff
@@ -1824,6 +1824,164 @@ VKTRACER_EXPORT VKAPI_ATTR VkResult VKAPI_CALL __HOOKED_vkAllocateDescriptorSets
...
+    uint32_t binding_index = 0;
+    trim::ObjectInfo* pInfo = trim::get_DescriptorSet_objectInfo(dstSet);
+    for (uint32_t i = 0; i < pInfo->ObjectInfo.DescriptorSet.numBindings; i++) {
+        if (binding == pInfo->ObjectInfo.DescriptorSet.pWriteDescriptorSets[i].dstBinding) {
+            binding_index = i;
+            break;
+        }
+    }
+    return binding_index;
+}
```
Comment (davidlunarg) [inline vktrace/vktrace_layer/vktrace_lib_trace.cpp]: This function will return 0 if binding is not found in the descriptor set. 0 can also be a valid return value when binding is found. Is this what you want?

Comment (mizhen) [reply vktrace/vktrace_layer/vktrace_lib_trace.cpp]: yes, that mean the target app has error or something wrong with trim descriptorset tracking. now will add a commit to add assert fail and let trim ignore the binding update for release build.

File: vktrace/vktrace_layer/vktrace_lib_trim.h [hunk 2]
```diff
@@ -90,6 +91,8 @@ enum enum_key_state {
 // return if hotkey triggered;
 bool is_hotkey_trim_triggered();
 
+static const uint32_t INVALID_BINDING_INDEX = std::numeric_limits<uint32_t>::max();
```
Comment (davidlunarg) [inline vktrace/vktrace_layer/vktrace_lib_trim.h]: Why is this defined in vktrace_lib_trim.h? It's not specific to trim functionality. Maybe it should be defined in vktrace_common.h?

**Comment by davidlunarg** [comment]
Merged the first two commits, please create a new pull request for the third commit.

**Pull request closed without merging**